### ISSUE TITLE
netflix show calendar display

### ISSUE THREAD

**Issue by wx2junior** [SchizoDuckie/DuckieTV#381]
When a Netflix only show exists on your calendar, all episodes are displayed.

proposed fix:

for Netflix show episodes which have the same date, aggregate them together, so instead of 

House of Cards (S03E01 3:01) on 02/27/15
House of Cards (S03E07 3:01) on 02/27/15
ETC... 13 times on one date

you would have

House of Cards (S03 13 New Episodes) on 02/27/15
one entry, representing the season and how many episodes.

this would prevent an issue if Netflix ever changed their model and released shows on a weekly basis. but would also alert you to the presence of new seasons being released.

**Comment by SchizoDuckie** [comment]
Hmmm I don't really feel like grouping shows like this. You can easily miss episodes this way and it's an exception just for netflix. If netflix changes their model, the code that handles this would also need adjusting. (countering what you say here) 

Right now it's just the same code handling everything.

**Comment by wx2junior** [comment]
Agreed one size fits all is the best approach, but if they release 13 episodes on one day, should it not be maybe a roll down + to extend the show and list all episodes for that day?

Or is it better to be like today and expand the calendar so far down that you need to scroll in order to see the end of the month?

If the answer is the second approach then it appears on the screen to the user as a bug from a UI/UX point of view.

**Comment by SchizoDuckie** [comment]
Well, if you think your calendar is too large we have week mode for the calendar (hit the little arrow-up below the current month :) 

Needing to scroll down is all based on screen resolution and how filled your calendar is ofcourse. I guess we could add an '<x more>' feature that shows by default, and that expands when you click it to keep the calendar clean though. It needs to be a user preference though, and you'd be the case to implement it for. All in all, i'd like some feedback from other people on this if they think this is a problem.

**Comment by wx2junior** [comment]
If anyone would like to see an example of the current layout, either add "House of Cards" to your local library or "Orange Is the New Black". I am adding screenshots for reference if anyone does not like those shows.

Also, your extension is extremely awesome.

**Comment by wx2junior** [comment]
![image](https://cloud.githubusercontent.com/assets/3940979/6531631/9d525562-c408-11e4-9fcb-d784ee1ebc61.png)

![image](https://cloud.githubusercontent.com/assets/3940979/6531636/a5b61fe0-c408-11e4-9c65-90995dfcb8ad.png)

**Comment by wx2junior** [comment]
I do realize that these two shows or any that are released in this fashion are the exception now, but I wouldn't be surprised if that changes over time. And I add them to my favorites for two reasons, I see when they come out and I get all the amazing artwork whenever I begin a browsing session.

**Comment by garfield69** [comment]
+1 for schizoduckie's suggested solution:
>add a '{show x more}' feature that shows a limited number of episodes by default, and that expands when you click it to keep the calendar clean.
>It needs to be a user preference setting that's disabled by default
I have 3 netflix and an amazon series that regularly make my calendar look untidy.

**Comment by wx2junior** [comment]
I must confess that I took a zip of this code and tried to read md's for 4 minutes between lunch and day job but couldn't immediately figure out where this change would occur...

Before anything is actually implemented, could you point me towards the direction of the code which would change (I assume in the controller part of an MVC setup but I know not enough about web or ruby)

I'd like to contribute to a solution if possible, if not I will learn and be greateful.

**Comment by Js41637** [comment]
Well the events for each day are just 
```
<calendar-event serie="event.serie" episode="event.episode" ng-if="hasEvent(day)" ng-repeat="event in getEvents(day)|filter:filterSpecials"></calendar-event>
```
so calendar-event repeated for each episode, a basic show more function is quite easy, just adding `limitTo:number` to the element however that isn't what we're looking for here, we want to limit a specific serie so I am guessing we will have to add a check in the functions that add events to the calendar.

**Comment by Js41637** [comment]
I don't think it's worth trying to accommodate for this at this point in time considering it's a very small minority of shows that do this (Netflix?) I do have this issue as well but that is where the 'Hide show on Calendar' option comes in. It's not really a solution but seeing as it all airs on the same day, once you watch or download or just know WHEN it aires, it isn't really needed on the calendar anymore like a conventional show that airs weekly and at possibly different times and periods of breaks. So hiding it works fine I think.

**Comment by garfield69** [comment]
I think that rather than focusing on it being Netflix or Amazon or whatever, a **SHOW MORE** option that can be set to limit the default max number of events for all the calendar day boxes, but which then allows you to see the rest on a button click, is worth pursuing. This would maintain the full calendar size by default to within the limits of the users screen size, instead of allowing a full box to cause the whole calendar to overflow anytime that days events exceed say, 5 events.
Once again I quote schizoduckie:
>add a '{show x more}' feature that shows a limited number of episodes by default, and that expands when you click it to keep the calendar clean.
>It needs to be a user preference setting that's disabled by default
I like this idea.

**Comment by bkd69** [comment]
The feature request I was about to post before reading this, is offering the option to customize the calendar scheduling of any given show, as well as setting an arbitrary season+episode start point, for catching up with historical seasons, and which would also work for binge dumps of stream service shows.

For example, Person of Interest is in its third season, but since I want to catch up on it, I should like to be able to tell DuckieTV to start with S1E05, and schedule it for every Tuesday and Thursday until we're all caught up. Likewise for a binge dump, like The Unbreakable Kimmy Schmidt, I could tell it to just schedule it for every Sunday, and it would start at episode 1, and increment it from there.

Should I enter this request as separate issue?

**Comment by Js41637** [comment]
Looked into @SchizoDuckie idea using something like
`ng-repeat="event in calendar.events " ng-show="$prev.serie.ID_Serie == event.serie.ID_Serie" ` 
Although I checked ng-repeat docs and it makes no mention of a $prev or $next function. I was able to get the next elements scope using normal JS `thingy.next().scope().event.serie.ID_Serie` although, dunno if it's just me but chrome was telling me that .prev() function doesn't even exist even though it apparently does from what I could search. I think what also makes things harder is how the ng-repeat is populated `event in getEvents(day)` it's not exactly an array of episodes we can track by $index.

**Comment by garfield69** [comment]
@js41637 read https://github.com/angular/angular.js/pull/9795
  ----- Original Message -----

**Comment by Js41637** [comment]
Yeah so it's not actually in angular?

**Comment by garfield69** [comment]
it has been added https://github.com/kwypchlo/angular.js/commit/e3c5a31ed3adada63282146c7ba1bd025a34acc4 but I'm not sure if it has made it out the to version that we are using...haven't had time to check.
  ----- Original Message -----

**Comment by Js41637** [comment]
Well the pull request hasn't been merged yet so I'm assuming it still hasn't been added and it's not in the docs.

**Comment by garfield69** [comment]
I guess that SchizoDuckie jumped the gun then ;-)
  ----- Original Message -----

**Comment by garfield69** [comment]
issue that needs a solution:
```html
          <td ng-if="getEventLimit(day)!=0" ng-repeat="day in week" class="day">
            <calendar-event serie="event.serie" episode="event.episode" ng-if="hasEvent(day)" ng-repeat="event in getEvents(day)|filter: filterSpecials|limitTo: getEventLimit(day) as eventCount"></calendar-event>
            <span ng-class="{'now':isNow(day),'active':isSameDay(day),'disabled':(day.getMonth()!=date.getMonth()),'after':isAfter(day),'before':isBefore(day)}" ng-bind="day.getDate()">
            </span>
            <button ng-if="eventCount.length==getEventLimit(day)" ng-click="bumpEventLimit(day)" class="btn btn-info" style="padding:0"><i class="glyphicon glyphicon-plus"></i>{{'SERIESLIST/TRAKT-TRENDING/show-more/btn'|translate}}</button>
          </td>
```
The ideal code would prevent the displaying the show-more button if the number of events passing through the ng-repeat filter was greater that the value set by the LimitTo.
The problem I have with the current code is that EventCount.length can only ever be less or equal to LimitTo.
This leads to the situation where, for example, 6 events passed the filter, and a limitTo of 6 causes the button to appear unnecessarily.
Haven't found a solution for this yet.

**Comment by SchizoDuckie** [comment]
Allright, so time for me to get into this i believe. I'm not really in favour of @garfield69's crude limit implementation, so I refactored the structure a bit to add series to an internal array of series, under which episodes are grouped.

Still need to fix the sorting, but I think this is a bit more solid, since with a random limit filter you may miss important shows.

**Comment by garfield69** [comment]
the  datepicker.html change in ec9b445ada37254ca3b16c7299b04fcf6afe79c8 fixes  ba7730a5bef01d4771a55e60b8e268c63a4d3f65 and thus completes this issue. Closed.

**Comment by garfield69** [comment]
oops, closed too soon. todo:
- [ ] sort events as per week view
- [ ] hide-series and hide-specials filter

**Comment by SchizoDuckie** [comment]
Some more styling and w.i.p. on this:

![image](https://cloud.githubusercontent.com/assets/111710/7055560/9c33b1d4-de44-11e4-9978-d866881b7832.png)

![image](https://cloud.githubusercontent.com/assets/111710/7055578/bd6ee8e6-de44-11e4-9bc8-44207548a9ab.png)

**Comment by garfield69** [comment]
Gorgeous!


  _____

**Comment by wx2junior** [comment]
I was just about to say that is beautiful! Great job!

**Comment by garfield69** [comment]
whats needed to prevent this?
![image1](https://cloud.githubusercontent.com/assets/6933240/7083158/a7e1b854-dfb2-11e4-85e6-4cce6316d1e3.jpg)

**Comment by SchizoDuckie** [comment]
Yea i thought that was going to happen. Its going to need some logic to
place the tooltip on the left for elements on the last day of the week

**Comment by garfield69** [comment]
I notice that it does not happen for those events on the last day of the week that have been marked as watched. Does that simplify things any?
  ----- Original Message -----

**Comment by garfield69** [comment]
![image](https://cloud.githubusercontent.com/assets/6933240/7083437/e2df69e4-dfb5-11e4-980c-98254b66dc82.png)

**Comment by Js41637** [comment]
Haven't looked at the code but the tooltips obviously display at the end/right width of the element, Marked as Watched episodes are obviously slightly smaller to accommodate for the tick. All that is needed to fix is this just change the containers for the name of show and add the tooltip to it or something similar, might be solvable with CSS but I don't have time to look at it atm.

**Comment by SchizoDuckie** [comment]
Last issue on this fixed with 5724af19763767d6827c404d3fcd3e3614617fc9
